problem-spring-common/src/main/java/org/zalando/problem/spring/common/AdviceTrait.java

```diff
@@ -0,0 +1,132 @@
+package org.zalando.problem.spring.common;
+
+import org.apiguardian.api.API;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.zalando.problem.*;
+
+import javax.annotation.Nullable;
+import java.net.URI;
+import java.util.Optional;
+
+import static java.util.Arrays.asList;
+import static org.apiguardian.api.API.Status.INTERNAL;
+import static org.apiguardian.api.API.Status.MAINTAINED;
+import static org.springframework.core.annotation.AnnotatedElementUtils.findMergedAnnotation;
+import static org.zalando.problem.spring.common.Lists.lengthOfTrailingPartialSubList;
+import static org.zalando.problem.spring.common.MediaTypes.PROBLEM;
+
+/**
+ * <p>
+ * Advice traits are simple interfaces that provide a single method with a default
+ * implementation. They are used to provide {@link ExceptionHandler} implementations to be used in
+ * Spring Controllers and/or in a {@link ControllerAdvice}. Clients can choose which traits they what to
+ * use à la carte.
+ * </p>
+ * <p>
+ * Advice traits are grouped in packages, based on they use cases. Every package has a composite advice trait that
+ * bundles all traits of that package.
+ * </p>
+ *
+ * @see ControllerAdvice
+ * @see ExceptionHandler
+ * @see Throwable
+ * @see Exception
+ * @see Problem
+ */
+@API(status = INTERNAL)
+public interface AdviceTrait {
+
+    Logger LOG = LoggerFactory.getLogger(AdviceTrait.class);
+
+    static void log(
```

Comment:
Even though these are static now exposes them still to clients. Ideally we should only *see* them in our *mvc* and * webflux*  modules, but nowhere else.

Reply:
Either something like [this](https://dzone.com/articles/java-8-declare-private-and-protected-methods-in-in) or we rely on the correctly declared API status.

Reply:
We could move all static methods into a separate top-level class called `AdviceTraits` (status internal) and only leave those (non-static) methods in here that are actually needed for both (mvc/webflux) implementations.

Reply:
Since the static method can only be called on the class and not the instance, I wouldn’t say that it is exposed. That said I can certainly move it to a distinct utility class. I have a AdviceUtils class in the other PR but I can also name it AdviceTraits. Tell me.

Reply:
Unlike for classes, interface static methods are not inherited: you need to call them from the "superinterface" and hence you don't get them in IDE autocompletion.
Anyway, I'll do the AdviceTraits class, no big deal 😄

Reply:
Done